src/main/java/ratpack/zipkin/internal/ZipkinHttpClientImpl.java

```diff
@@ -105,14 +97,64 @@ public void close() {
         return request(uri, requestConfigurer.prepend(RequestSpec::post));
     }
 
-    private Action<? super RequestSpec> tracedRequestAction(final Action<? super RequestSpec> action) {
-        return action.append(requestSpec -> {
-            MethodCapturingRequestSpec captor = new MethodCapturingRequestSpec(requestSpec);
-            action.execute(captor);
-            HttpMethod capturedMethod = Optional.ofNullable(captor.getCapturedMethod()).orElse(HttpMethod.GET);
-            requestInterceptor
-                    .handle(requestAdapterFactory.createAdaptor(requestSpec.method(capturedMethod), capturedMethod.getName()));
-        });
+    private void tracedResponse(Result<?> result, StatusCode statusCode, @Nullable Span span) {
+        if (span != null) {
+            if (result.isError()) {
+                span.tag(Constants.ERROR, result.getThrowable().getMessage());
+            } else {
+                int status = statusCode.getStatusCode();
+                if (status < 200 || status > 399) {
+                    span.tag(TraceKeys.HTTP_STATUS_CODE, String.valueOf(status));
+                } else if (status > 499) {
+                    span.tag(Constants.ERROR, "server error " + status);
+                }
+            }
+
+            span.finish();
+        }
+    }
+
+    private interface StatusCode {
+        int getStatusCode();
+    }
+
+    private void tracedStreamedResponse(Result<StreamedResponse> response, @Nullable Span span) {
+        tracedResponse(response, () -> response.getValue().getStatusCode(), span);
+    }
+
+    private void tracedResponse(Result<ReceivedResponse> response, @Nullable Span span) {
+        tracedResponse(response, () -> response.getValue().getStatusCode(), span);
+    }
+
+    static class ExecuteWithSpan {
+
+        final Tracer tracer;
+        Span span;
+
+        ExecuteWithSpan(Tracer tracer) {
+            this.tracer = tracer;
+        }
+
+        Action<? super RequestSpec> execute(Action<? super RequestSpec> action) {
+            return (request -> {
+                Span parent = tracer.currentSpan();
```

Comment:
Thanks for the suggestion.  It wasn't clear to me if this would start a new trace if one wasn't already in progress.  I will switch to `tracer.nextSpan()` instead.